docs/source/opens-upgrade.rst

```diff
@@ -0,0 +1,133 @@
+
+This document details the procedure for upgrading a Calico-based OpenStack
+system. It contains the full steps for upgrading all components and the order
+in which that upgrade should be performed. Most releases do not concurrently
+upgrade all of these components: if a release does not upgrade a given
+component, you may skip those steps.
+
+.. warning:: While the upgrade procedure is very safe, you will be unable to
+             issue API requests to your OpenStack system during the procedure.
+             Please plan your upgrade window accordingly, and see the
+             :ref:`service_impact` section for more details.
+
+
+.. _service_impact:
+
+Service Impact
+--------------
+
+During the upgrade, **all VMs will continue to function normally**: there
+should be no impact on the data plane. However, control plane traffic may fail
+at different points throughout the upgrade.
+
+Generally, users should be prevented from creating or updating virtual machines
+during this procedure, as these actions will fail. VM deletion *may* succeed,
+but will likely be delayed until the end of the upgrade.
+
+For this reason, we highly recommend planning a maintenance window for the
+upgrade. During this window, you should disable all user API access to your
+OpenStack deployment.
+
+Upgrade Procedure
+-----------------
+
+Upgrade is performed in the following stages.
+
+1: Upgrade etcd
+~~~~~~~~~~~~~~~
+
+This step should be run on every machine in your deployment that runs any
+Calico code, and also on the machine running the etcd cluster.
+
+Ubuntu 14.04
+^^^^^^^^^^^^
+
+On the machine running etcd, simply use apt-get to obtain the more recent
+version::
+
+    apt-get update
+    apt-get install etcd
+
+Red Hat 7
+^^^^^^^^^
+
+On the machine running etcd, begin by stopping the etcd process::
+
+    systemctl stop etcd
+
+Then, download the latest binary and install it::
+
+    curl -L  https://github.com/coreos/etcd/releases/download/v2.0.11/etcd-v2.0.11-linux-amd64.tar.gz -o etcd-v2.0.11-linux-amd64.tar.gz
+    tar xvf etcd-v2.0.11-linux-amd64.tar.gz
+    cd etcd-v2.0.11-linux-amd64
+    mv etcd* /usr/local/bin/
+
+Now, restart etcd::
+
+    systemctl start etcd
+
+2: Upgrade compute software
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+On each machine running the Calico compute software (the component called
+Felix), run the following upgrade steps.
+
+Ubuntu 14.04
+^^^^^^^^^^^^
+
+First, upgrade packaged components::
+
+    apt-get update
+    apt-get install dnsmasq-base nova-api-metadata neutron-dhcp-agent neutron-common python-etcd calico-compute
+
+Finally, restart Felix to ensure that it picks up any changes::
+
+    service calico-felix restart
+
+Red Hat 7
+^^^^^^^^^
+
+First, upgrade python-etcd::
+
+    curl -L https://github.com/Metaswitch/python-etcd/archive/master.tar.gz -o python-etcd.tar.gz
+    tar xvf python-etcd.tar.gz
+    cd python-etcd-master
+    python setup.py install
+
+Then, update the relevant components::
+
```

Comment:
Will yum update update other things beyond Calico?  Just thinking that in the Ubuntu instructions we're explicitly upgrading only the Calico stuff, but if yum update could upgrade other packages, that could have unintended consequences.  Is there a way to just update the Calico stuff?